### PR TITLE
[SPARK-23992][CORE] ShuffleDependency does not need to be deserialized every time

### PR DESCRIPTION
**Pull request by 10110346**
## What changes were proposed in this pull request?

In the same stage, 'ShuffleDependency' is not necessary to be deserialized each time.

## How was this patch tested?
Existing unit test

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala [hunk 1]
```diff
@@ -113,3 +118,24 @@ private[spark] class ShuffleMapTask(
 
   override def toString: String = "ShuffleMapTask(%d, %d)".format(stageId, partitionId)
 }
+
+object ShuffleMapTask extends Logging {
+  private val cache = CacheBuilder.newBuilder()
```
Comment (Ngone51) [inline core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala]: Do we need to clear this `cache` at the end of a app ?

Comment (10110346) [reply core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala]: we don't need to clear, this cache can  recovery itself

**Comment by jiangxb1987** [comment]
Have you seen any performance improvement with this change?

**Comment by HyukjinKwon** [comment]
ping @10110346 for ^

**Comment by 10110346** [comment]
I'm sorry to be so late reply to you!
I have tested in my production environment,  it has a bit of performance improvement. @jiangxb1987 @HyukjinKwon

**Comment by HyukjinKwon** [comment]
Mind if I ask to share the number and post it in the PR description?

**Pull request closed without merging**